### PR TITLE
Generate scipy api automatically

### LINKED ISSUE #58: Organize properties generation (improve `print_scipy_api_list.py`)
Here is my suggestion to generate a scipy API list in a sophisticated way.

# Code

Consider the following code:

```python
import inspect

from scipy import interpolate, optimize, spatial, stats

def print_apis(pymod):
    _, name = pymod.__name__.split(".")
    print(f"{name}_properties = [")
    spaces = 4 * " "
    attributes = sorted(set(pymod.__all__))
    for attr in attributes:
        condition = 0
        for target in ["isfunction", "isclass"]:
            b = eval(f"inspect.{target}(pymod.{attr})")
            condition += int(b)
        if condition > 0:
            print(spaces + ":" + attr + ",")
    print("]")
```

If we execute `print_apis(spatial)`, we get output as below:

```julia
spatial_properties = [
    :ConvexHull,
    :Delaunay,
    :HalfspaceIntersection,
    :KDTree,
    :QhullError,
    :Rectangle,
    :SphericalVoronoi,
    :Voronoi,
    :cKDTree,
    :convex_hull_plot_2d,
    :delaunay_plot_2d,
    :distance_matrix,
    :geometric_slerp,
    :minkowski_distance,
    :minkowski_distance_p,
    :procrustes,
    :voronoi_plot_2d,
]
```

The output above can be interpreted in Julia REPL as an instance of `Vector{Symbol}`

# Why we need

If we run the current https://github.com/AtsushiSakai/SciPy.jl/blob/master/src/print_scipy_api_list.py  we get (especially spatial module part)

```
=== spatial module ===
["ConvexHull", "Delaunay", "HalfspaceIntersection", "KDTree", "QhullError", "Rectangle", "SphericalVoronoi", "Voronoi", "cKDTree", "ckdtree", "convex_hull_plot_2d", "delaunay_plot_2d", "distance", "distance_matrix", "geometric_slerp", "kdtree", "minkowski_distance", "minkowski_distance_p", "procrustes", "qhull", "tsearch", "voronoi_plot_2d", "distance", "transform"]
```

The above includes ,for example, a literal `distance` which is a Python module not a function in Python actually (Ref https://docs.scipy.org/doc/scipy/reference/spatial.distance.html). Therefore it is better to treat `distance` as  a module rather than function. Note that `print_apis` just collects functions and classes via:

```python
...
        for target in ["isfunction", "isclass"]:
            b = eval(f"inspect.{target}(pymod.{attr})")
            condition += int(b)
...
```

Also `print_apis` displays scipy apis line by line which improves readability 👍

### PR DESCRIPTION
**Pull request by terasakisatoshi**
resolve https://github.com/AtsushiSakai/SciPy.jl/issues/58

From now we only have to write the following code:

```julia
pyimport_conda(“scipy”, “scipy”)
@pyinclude(joinpath(pkgdir(@__MODULE__), "src", "scipy_api_list.py"))
apis = py"generate_scipy_apis"("<submodule_name>")

all_properties = [apis["function"]; apis["class"]]
```

to get `all_properties` for each `<submodule_name>` e.g. `spatial`, `optimize` etc...

### CODE REVIEW

File: src/scipy_api_list.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+"""
+
+This is a python script to show API for each submodules of SciPy
+
+"""
+import importlib
+import inspect
+from collections import defaultdict
+
+import numpy
```
Comment (AtsushiSakai) [inline src/scipy_api_list.py:10]: [Question] Why we need to import numpy? Just checking numpy version?

**Review by AtsushiSakai — COMMENTED** [review]
Circle CI failure looks weird, but let's ignore it in this PR. 
All changes look great. I have one question.

File: src/scipy_api_list.py (hunk 1 above)
Comment (terasakisatoshi) [reply src/scipy_api_list.py:10]: > Just checking numpy version?

Yes, the current`print_scipy_api_list.py` in master branch still privides the version of numpy.

https://github.com/AtsushiSakai/SciPy.jl/blob/ddf4eab371cf47ee0a9550edd69a2d1c6770974f/src/print_scipy_api_list.py#L7

Do you use this functionality? If not, I'll remove lines depend on numpy e.g. `import numpy` and `numpy.__version__` etc...

Comment (AtsushiSakai) [reply src/scipy_api_list.py:10]: Ah, I forget it. Maybe  this is debug use when scipy has an issue because of numbly version. Let’s keep it

**Review by AtsushiSakai — APPROVED** [review]
LGTM. After fix #46, let’s release new version!!

**Comment by terasakisatoshi** [comment]
🚀

**Pull request merged**